v3-docs/v3-migration-docs/breaking-changes/call-x-callAsync.md

```diff
@@ -32,105 +28,209 @@ Meteor.call('getAllData', (err, res) => { // [!code highlight]
 });
 
 // render data or error
-
-
 ```
 
-
 ```js [v2-server.js]
-import { Meteor } from 'meteor/meteor'
-import { Mongo } from 'meteor/mongo'
+import { Meteor } from "meteor/meteor";
+import { Mongo } from "meteor/mongo";
 
-const MyCollection = new Mongo.Collection('myCollection');
+const MyCollection = new Mongo.Collection("myCollection");
 
 Meteor.methods({
   getAllData() {
     return MyCollection.find().fetch(); // [!code highlight]
-  }
+  },
 });
 ```
 
 ```js [v3-client.jsx]
-import { Meteor } from 'meteor/meteor'
+import { Meteor } from "meteor/meteor";
 
 try {
-  const data = await Meteor.callAsync('getAllData'); // [!code highlight]
+  const data = await Meteor.callAsync("getAllData"); // [!code highlight]
   // render data
 } catch (error) {
   // render error
 }
-
 ```
 
 ```js [v3-server.js]
-import { Meteor } from 'meteor/meteor'
-import { Mongo } from 'meteor/mongo'
+import { Meteor } from "meteor/meteor";
+import { Mongo } from "meteor/mongo";
 
-const MyCollection = new Mongo.Collection('myCollection');
+const MyCollection = new Mongo.Collection("myCollection");
 
 Meteor.methods({
   async getAllData() {
     return await MyCollection.find().fetchAsync(); // [!code highlight]
-  }
+  },
 });
-
 ```
 
 :::
 
 ## Rules of using `Meteor.callAsync` & `Meteor.call`
 
-::: tip 
-It is not recommended to use concurrent calls.
-Use `await` for your `Meteor.callAsync`.
-:::
+You can still use `Meteor.callAsync` and `Meteor.call` together. But there are some catches that you need to
+pay attention to. Let's go over them.
+
+### How to NOT use them together
 
-here are a few examples of cases where you should use `Meteor.callAsync` instead of `Meteor.call`:
+These are for **async** method when called with `Meteor.callAsync` or `Meteor.call`.
+
+**Case 1**: Calling `Meteor.callAsync` and then right after calling `Meteor.call` when the methods have stubs.
 
 ```js
-import { Meteor } from 'meteor/meteor'
+import { Meteor } from "meteor/meteor";
+
+Meteor.callAsync("SOME_ASYNC_METHOD_WITH_STUB")
+  .then((data) => console.log(data))
+  .catch((err) => console.log(err));
 
-Meteor.call("someMethod", (err, res) => { // [!code error] This is not ok
+Meteor.call("SOME_ASYNC_METHOD_WITH_STUB", (err, res) => {
```

Comment:
I'm assuming the stub used with `Meteor.call` is async. This seems to be broken due to the changes in `EV.withValue` to support promises on the client: https://github.com/meteor/meteor/blob/818c5da95ca5b884e61ad153ad27ccc7ec86741b/packages/meteor/dynamics_browser.js#L46-L52

Before, async stubs were not supported in `Meteor.apply`. With this change, they still aren't really supported, but it introduces the same issues we had with `Meteor.callAsync` in Meteor 2 that the queueing fixed.

 If we do want to support them, [this code](https://github.com/meteor/meteor/blob/818c5da95ca5b884e61ad153ad27ccc7ec86741b/packages/ddp-client/common/livedata_connection.js#L612-L620) would need to be updated, as well as how the queueing handles `Meteor.apply`.

Though I don't think we should support async stubs in `Meteor.apply`.